Analysis/src/EventTags.cpp

```diff
@@ -3,34 +3,71 @@ This file is part of https://github.com/hh-italian-group/hh-bbtautau. */
+    std::string complete_name;
+    if(unc_scale!= UncertaintyScale::Central && unc_scale!= UncertaintyScale::Up && unc_scale!= UncertaintyScale::Down)
+        throw exception("Incorrect uncertainty scale");
+
+    if(unc_source==UncertaintySource::None){
+        if(unc_scale==UncertaintyScale::Central)
+            complete_name="Central";
+        else
+            throw exception("Central Value can't have different scales");
+    }
+    else{
+        if(unc_scale!=UncertaintyScale::Central)
+            complete_name=ToString(unc_source)+"_"+ToString(unc_scale);
+        else{
+            complete_name=ToString(unc_source);
+        }
+    }
+    return complete_name;
+
+}*/
+
 
 EventTagCreator::EventTagCreator(const EventCategorySet& _categories, const EventSubCategorySet& _subCategories,
                                   const std::set<UncertaintySource>& _event_unc_sources,
                                   const std::set<UncertaintySource>& _norm_unc_sources,
-                                  bool _use_IterativeFit, Channel _channel, Period _period) :
+                                  bool _use_IterativeFit, std::string _json_file)  :
         categories(_categories), subCategories(_subCategories), event_unc_sources(_event_unc_sources),
-        norm_unc_sources(_norm_unc_sources), use_IterativeFit(_use_IterativeFit), channel(_channel), period(_period)
+        norm_unc_sources(_norm_unc_sources), use_IterativeFit(_use_IterativeFit),
+        json_file(_json_file)
 {
     if(use_IterativeFit) {
-        static const std::map<std::pair<Channel, Period>, float> iterativeFit_corrections = {
-            {{Channel::ETau, Period::Run2016}, 1.0120f}, // 1.01203715762
-            {{Channel::MuTau, Period::Run2016}, 1.013f}, // 1.01296331792
-            {{Channel::TauTau, Period::Run2016}, 1.0101f}, // 1.01010364517
-            {{Channel::ETau, Period::Run2017}, 0.9949f}, // 0.994930642536
-            {{Channel::MuTau, Period::Run2017}, 0.9993f}, // 0.999263715405
-            {{Channel::TauTau, Period::Run2017}, 0.9547f}, // 0.954725518543
-            {{Channel::ETau, Period::Run2018}, 1.0004f}, // 1.0003768284
-            {{Channel::MuTau, Period::Run2018}, 1.0039f}, // 1.00388888346
-            {{Channel::TauTau, Period::Run2018}, 0.9795f}, // 0.97949255088
-        };
-        auto iter = iterativeFit_corrections.find({channel, period});
-        if(iter == iterativeFit_corrections.end())
-            throw exception("Normalization correction for iterative fit is not available for %1% %2%")
-                    % period % channel;
-        iterativeFit_correction = iter->second;
+        std::vector<UncertaintyScale> unc_scales= {UncertaintyScale::Central, UncertaintyScale::Up, UncertaintyScale::Down};
+        boost::property_tree::ptree loadPtreeRoot;
+        boost::property_tree::read_json(json_file, loadPtreeRoot);
+        for(UncertaintySource unc_source:event_unc_sources){
```

Comment:
- why do you loop only for event_unc_sources and not norm?
- better to do other way around: loop over nodes available in json and fill the map